docs/lang_spec/Cadence Language Specification.md

```diff
@@ -0,0 +1,5798 @@
+      _ Example: `0x2A`, `0x3e8`.
+  - Underscores (`_`) can be used to separate digits for readability: `1_000`, `0x3E_8`.
+
+- **Fixed-Point Literals**:
+  - Fixed-point literals represent real numbers with a decimal point. **TODO** Need more information on fixed-point.
+    - Example: `3.14`, `1.23`.
+
+- **Floating-Point Literals**:
+  - Fixed-point is not supported by Cadence.
+
+- **String Literals**:
+  - String literals are sequences of characters enclosed in double quotes (`"`). 
+  - Strings support escape sequences for Unicode scalars and special characters.
+    - `\0`: Null character
+    - `\\`: Backslash
+    - `\t`: Horizontal tab
+    - `\n`: Line feed
+    - `\r`: Carriage return
+    - `\"`: Double quotation mark
+    - `\'`: Single quotation mark
+    - `\u`: A Unicode scalar value, written as `\u{x}`, where `x` is a 1–8 digit hexadecimal number which needs to be a valid Unicode scalar value, i.e., in the range 0 to 0xD7FF and 0xE000 to 0x10FFFF inclusive
+    - Example: `"Hello, Cadence\u{21}"`, `"Multiline\nString"`.
+
+- **Boolean Literals**:
+  - Boolean literals represent the two truth values: `true` and `false`.
+
+- **Special Literals**:
+  - **Nil Literal**: The literal `nil` represents the absence of a value in Cadence’s optional types.
+  - **Resource Literals**: Resources are created and destroyed using specific syntax in Cadence, but the core value does not have a literal representation. Instead, resources are referenced through special constructs like `create` and `destroy`.
+
+- **Array Literals**:
+  - **TODO**
+
+- **Dictionary Literals**:
+  - **TODO**
+
+---
+
+### 5. Comments
+
+Comments in Cadence are non-executable sections of the code that can be used for documentation, explanation, or temporarily disabling code. Cadence supports two types of comments:
+
+- **Single-line Comments**:
+  - Single-line comments start with `//` and continue until the end of the line.
+  - Example: 
+    ```cadence
+    // This is a single-line comment
+    let value = 42  // This is also a single-line comment
+    ```
+
+- **Multi-line Comments**:
+  - Multi-line comments start with `/*` and end with `*/`. They can span multiple lines and can be nested.
+  - Example:
+    ```cadence
+    /* This is a 
+       multi-line comment */
+    ```
+
+  - **Nested Comments**:
+    - Cadence allows multi-line comments to be nested, which means you can comment out blocks of code that contain multi-line comments without causing errors.
+    - Example:
+      ```cadence
+      /* This is a multi-line comment 
+         /* This is a nested comment */ 
+         Ending the outer comment */
+      ```
+
+---
+
+This lexical structure ensures a clear and consistent foundation for writing Cadence programs, supporting both readability and maintainability across a variety of use cases within the Flow blockchain.
+
+## III. Syntax and Grammar
```

Comment:
If we structure the specification in terms of layers, like all features in terms of syntax, than all features in terms of semantics, etc. it might make sense to explain the purpose of a feature in one section. 

For example, when describing the syntax for a struct declaration, we could  only focus on the syntax, and defer the explanation and purpose for the feature to another section (semantics?), e.g. "the syntax ... introduces a struct declaration", where "struct declaration" links to the section explaining the feature and semantics for structs.